towhee/utils/git_utils.py

```diff
@@ -0,0 +1,191 @@
+# Copyright 2021 Zilliz. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import subprocess
+from pathlib import Path
+from typing import Union, List
+import sys
+
+# TODO
+# from towhee.utils.hub_utils import HubUtils
+from towhee.utils.log import engine_log
+
+
+class GitUtils:
+    """
+    A wrapper class to wrap git manipulations.
+
+    Args:
+        author (`str`):
+            The author of the repo.
+        repo (`str`):
+            The name of the repo.
+        root (`str`):
+            The root url of the repo.
+        self._repo_path (`Union[str, Path]`):
+            The path to the local repo.
+    """
+    def __init__(self, author: str, repo: str, root: str = 'https://hub.towhee.io', repo_path: Union[str, Path] = None):
```

Comment:
The `repo_path` maybe named `local_repo_path`

Reply:
Overview of the code, it seems only `clone` needs `repo_path`

Reply:
Fixed